### ISSUE TITLE
"xgboost.core.XGBoostError: Required parameter max_delta_step of float is not presented"

### ISSUE THREAD

**Issue by iosusan** [dmlc/xgboost#2662]
I'm getting this error when trying to load a count:poisson trained model into python.

I've trained a simple model from pima-indians dataset ( https://archive.ics.uci.edu/ml/datasets/pima+indians+diabetes ). 

If i use xgb.XGBRegressor() i can pickle.dump it and load it afterwards and effectively use it.
But if i use xgb.XGBRegressor(objective="count:poison") i cannot load the pickled model because "Required parameter max_delta_step of float is not presented"

(I got the same results saving an "count:poison" model from R then trying to load it from python, but i supposed it could be a library mismatch, this is not the point of this issue)

## Environment info
Operating System: ubuntu 16.04

Compiler: gcc (Ubuntu 5.4.0-6ubuntu1~16.04.4) 5.4.0 20160609

Package used (python/R/jvm/C++): xgboost 0.6a2 from pip

`xgboost` version used: 0.6a2

If you are using python package, please provide

1. The python version and distribution: tried both 2.7.12 and 3.5.2
2. The command to install `xgboost` if you are not installing from source: pip install xgboost

## Steps to reproduce

1. create an XGBRegressor without objective function, and train it ( i guess then it takes a reg:linear) and pickle.dump it, then you can load it (pickle.load(open('myfile'))
2. add the parameter objective="count:poison" to the XGBRegressor constructor, train it, pickle.dump it - when trying to load_model i get "xgboost.core.XGBoostError: Required parameter max_delta_step of float is not presented"
3. an example code is here, just switch between the both model = xgb.XGBRegressor lines to see both behaviours:

```
from numpy import loadtxt
import xgboost as xgb
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score
import pickle

dataset = loadtxt('pima-indians-diabetes.csv', delimiter=',')

X = dataset[:,0:8]
Y = dataset[:,8]

seed = 7
test_size = 0.33
X_train, X_test, y_train, y_test = train_test_split(X,
                                                    Y,
                                                    test_size=test_size,
                                                    random_state=seed)

model = xgb.XGBRegressor(objective='count:poisson')
#model = xgb.XGBRegressor()
model.fit(X_train, y_train)

print(model)

y_pred = model.predict(X_test)
predictions = [round(value) for value in y_pred]

accuracy = accuracy_score(y_test, predictions)
print("Accuracy %.2f%%" % (accuracy*100.0))

# now save model t a binary file
pickle.dump(model, open('pima_test_model.bin', 'wb'))

# and load
loaded_model = pickle.load(open('pima_test_model.bin', 'rb'))

```

## What have you tried?

1. several ways to pass that parameter to constructor, booster, etc.. when loading the model but all of them are ignored
2. different objective functions seem to work such as reg:linear
3. different python versions 2.7 and 3.5 give the same error

**Comment by khotilov** [comment]
This was fixed in #1615. The pypi package is too old.
In future, please use github search before creating a new issue.

**Comment by iosusan** [comment]
@khotilov you're right, srry! i initially thought it was more related to max_delta_step param than to the poisson objective function itself, i only realized it was poisson when filling in the bug report.

thanks!
